packages/expect/src/__tests__/matchers.property.test.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+const chalk = require('chalk');
+const chalkEnabled = chalk.enabled;
+const fc = require('fast-check');
+
+// settings for anything arbitrary
+const anythingSettings = {
+  withBoxedValues: true,
+  withMap: false,
+  withSet: false,
+};
+
+// assertion settings
+const assertSettings = {}; // eg.: {numRuns: 10000}
+
+beforeAll(() => {
+  chalk.enabled = true;
+});
+
+afterAll(() => {
+  chalk.enabled = chalkEnabled;
+});
+
+describe('.toStrictEqual()', () => {
+  it('should distinguish distinct values', () => {
+    fc.assert(
+      fc.property(
+        fc.anything(anythingSettings),
+        fc.anything(anythingSettings),
+        (a, b) => {
+          // Given:  a and b values such as a different from b
+          // Assert: We expect `expect(a).not.toStrictEqual(b)`
+          const replacer = (k, v) => {
+            if (v === undefined) return '<undefined>';
+            if (v instanceof Set)
+              return `new Set([${JSON.stringify(Array.from(v), replacer)}])`;
+            if (v instanceof Map)
+              return `new Map([${JSON.stringify(Array.from(v), replacer)}])`;
+            return v;
+          };
+          fc.pre(JSON.stringify(a, replacer) !== JSON.stringify(b, replacer));
+          expect(a).not.toStrictEqual(b);
+        },
+      ),
+      assertSettings,
+    );
+  });
+
+  it('should be reflexive', () => {
+    fc.assert(
+      fc.property(fc.dedup(fc.anything(anythingSettings), 2), ([a, b]) => {
+        // Given:  a and b identical values
+        // Assert: We expect `expect(a).toStrictEqual(b)`
+        expect(a).toStrictEqual(b);
+      }),
+      assertSettings,
+    );
+  });
+
+  it('should be symmetric', () => {
+    const safeExpectStrictEqual = (a, b) => {

```

Comment:
The wrapper is needed because I want to assess that `expect(a).toEqual(b)` and `expect(b).toEqual(a)` are equivalent for both success case and failure case.
It is used inside: `expect(safeExpectStrictEqual(a, b)).toBe(safeExpectStrictEqual(b, a))`.

The test might produce values such as `a == b` but also values such as `a != b` - which will throw.